etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiMessageConverterHelperTest.groovy

```diff
@@ -0,0 +1,97 @@
+package gov.hhs.cdc.trustedintermediary.external.hapi
+
+
+import gov.hhs.cdc.trustedintermediary.context.TestApplicationContext
+import org.hl7.fhir.r4.model.*
+import spock.lang.Specification
+
+class HapiMessageConverterHelperTest extends Specification {
+
+    Bundle mockBundle
+    Patient mockPatient
+
+    def setup() {
+        TestApplicationContext.reset()
+        TestApplicationContext.init()
+        TestApplicationContext.register(HapiMessageConverterHelper, HapiMessageConverterHelper.getInstance())
+        TestApplicationContext.injectRegisteredImplementations()
+
+        mockPatient = new Patient()
+        mockBundle = new Bundle().addEntry(new Bundle.BundleEntryComponent().setResource(mockPatient))
+    }
+
+    def "addEtorTag adds the ETOR message header tag to any Bundle"() {
+        given:
+        def expectedSystem = "http://localcodes.org/ETOR"
+        def expectedCode = "ETOR"
+        def expectedDisplay = "Processed by ETOR"
+
+        def messageHeader = new MessageHeader()
+        messageHeader.setId(UUID.randomUUID().toString())
+        def messageHeaderEntry = new Bundle.BundleEntryComponent().setResource(messageHeader)
+        mockBundle.getEntry().add(messageHeaderEntry)
```

Comment:
Given you're modifying `mockBundle` here, which is defined as a global variable, should it be of local scope? Will changing `mockBundle` affect other tests?

Reply:
The `setUp()` method takes care of this. Spock runs `setup()` before each individual test to ensure we get new variables with each test.

Reply:
Good question. I'll check into it and follow up on it.

Reply:
I added `mockBundle = null` as part of the setup(). This will ensure the mockBundle gets nulled out from the start, every time setup() runs.